server/model/config.go

```diff
@@ -3751,6 +3752,16 @@
 		return NewAppError("Config.IsValid", "model.config.is_valid.collapsed_threads.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	// we check if file has a valid parent, the server will try to create the socket
+	// file if it doesn't exist, but we need to be sure if the directory exist or not
+	if *s.EnableLocalMode {
+		parent := filepath.Dir(*s.LocalModeSocketLocation)
+		_, err := os.Stat(parent)
```

Comment:
Should we engage Security on this (e.g. does it need some form of validation)? I suppose it may be fine given it's an admin only config setting but better to be thorough.